src/mscorlib/corefx/System/Globalization/CompareInfo.cs

```diff
@@ -379,13 +379,17 @@ public unsafe virtual bool IsPrefix(String source, String prefix, CompareOptions
                     SR.ArgumentNull_String);
             }
             Contract.EndContractBlock();
-            int prefixLen = prefix.Length;
 
-            if (prefixLen == 0)
+            if (prefix.Length == 0)
             {
                 return (true);
             }
 
+            if (source.Length == 0)
```

Comment:
This one is interesting because in non ordinal cases we may get prefix with all undefined characters which will be evaluated as equal to null string (I guess). 

I am seeing @ellismg is the one applied this change 

1443498 edit Matt Ellis 3/31/2015 6:51:58 PM $/DevDiv/FX/Product/ProjectN/ndp/FxCore/src/System.Private.CoreLib/System/Globalization/CompareInfo.cs DivDiv #1142743: [TestCop] System.Runtime.Numerics test are failing for ProjectN and ProjectNRel While re-factoring glob...

Matt may have more info about that but I suggest to include this one till Matt is back and confirm it